helix-term/src/commands.rs

```diff
@@ -4240,3 +4355,246 @@ fn suspend(_cx: &mut Context) {
     #[cfg(not(windows))]
     signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
 }
+
+// DAP
+fn dap_start(cx: &mut Context) {
+    use helix_dap::Client;
+    use helix_lsp::block_on;
+    use serde_json::to_value;
+
+    let (_, doc) = current!(cx.editor);
+
+    // TODO config picker
+    let path = match doc.path() {
+        Some(path) => path.to_path_buf(),
+        None => {
+            cx.editor
+                .set_error("Can't start debug: document has no path".to_string());
+            return;
+        }
+    };
+
+    let config = cx
+        .editor
+        .syn_loader
+        .language_config_for_file_name(&path)
+        .and_then(|x| x.debug_adapter.clone());
+    let config = match config {
+        Some(c) => c,
+        None => {
+            cx.editor.set_error(
+                "Can't start debug: no debug adapter available for language".to_string(),
+            );
+            return;
+        }
+    };
```

Comment:
You don't need to do this lookup, the document already contains the config: `doc.language_config()`

Reply:
cool, thanks